### PR TITLE
New Plugin: Bosswerk micro inverters

### PR DESCRIPTION
**Pull request by mzanetti**
nymea-plugins pull request checklist:

- [x] Make sure the pull request's title is of format "Plugin name: Add support for xyz" or "New plugin: Plugin name"

- [x] Did you test the changes on hardware, if not (e.g. absence of required hardware), please mention a person to confirm it has been tested.

- [x] Did you update the plugin's README.md accordingly?

- [x] Did you update translations (`cd builddir && make lupdate`)?

### CODE REVIEW

File: bosswerk/integrationpluginbosswerk.cpp [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+    connect(monitor, &NetworkDeviceMonitor::reachableChanged, thing, [timer, thing](bool reachable) {
+        thing->setStateValue("connected", reachable);
+        if (reachable) {
+            timer->start();
+        } else {
+            timer->stop();
+        }
+    });
+
+    connect(timer, &PluginTimer::timeout, this, [this, thing](){
```
Comment (t-mon) [inline bosswerk/integrationpluginbosswerk.cpp:152]: `connect(timer, &PluginTimer::timeout, thing, `

File: bosswerk/integrationpluginbosswerk.cpp [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    QString username = pluginStorage()->value("username").toString();
+    QString password = pluginStorage()->value("password").toString();
+    pluginStorage()->endGroup();
+
+    QUrl url("http://" + username + ":" + password + "@" + monitor->networkDeviceInfo().address().toString() + "/status.html");
+    qCDebug(dcBosswerk()) << "Requesting" << url.toString();
+    QNetworkRequest request(url);
+
+    QNetworkReply *statusReply = hardwareManager()->networkManager()->get(request);
+    connect(statusReply, &QNetworkReply::finished, thing, [=](){
```
Comment (t-mon) [inline bosswerk/integrationpluginbosswerk.cpp:183]: `connect(statusReply, &QNetworkReply::finished, statusReply, &QNetworkReply::deleteLater);`

File: bosswerk/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+* MI-300
+* MI-600
+
+## Requirements
+
+The solar inverter needs to be connected to the same network as nymea. Once powered on, the
+inverter will open a wireless hotspot named AP_XXXXXXXXXX where XXXXXXXXXX is the serial number
+written on the casing. The default password for this WiFi is 12345678. After connecting,
+open http://10.10.100.254 with the browser and use the web interface on the inverter to connect
```
Comment (t-mon) [inline bosswerk/README.md:15]: maybe use `[http://10.10.100.254](http://10.10.100.254)`, so it will be a link you can click on right the way...

**Review by t-mon — CHANGES_REQUESTED** [review]

File: bosswerk/integrationpluginbosswerk.cpp (hunk 1 above)
Comment (mzanetti) [reply bosswerk/integrationpluginbosswerk.cpp]: good catches! thanks. Fixed

**Review by t-mon — APPROVED** [review]

**Pull request merged**